Documentation/ExtensionArchitecture/ComposerJson/Index.rst

```diff
@@ -9,13 +9,14 @@
 composer.json
 =============
 
-*-- required*
+*-- required* in composer-based installations
 
 .. note::
 
    While the file :file:`composer.json` is currently not strictly required
-   for an extension to function properly, it is considered
-   bad practice not to add one. That is why we classify it as "required".
+   for an extension to function properly in legacy non-Composer installations
```

Comment:
```suggestion
   for an extension to function properly in Legacy installations
```

Reply:
I would keep this as it makes more clear what kind of installation is talked about